pkg/cluster/networkprofile.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/feature"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PopulateMTUSize ensures that every new cluster object has the MTUSize field defined
+func (m *manager) populateMTUSize(ctx context.Context) error {
+	// Get appropriate MTU size
+	mtuSize := api.MTU1500
+	subProperties := m.subscriptionDoc.Subscription.Properties
+	if feature.IsRegisteredForFeature(subProperties, api.FeatureFlagMTU3900) {
+		mtuSize = api.MTU3900
+	}
+
+	// Patch the cluster object with correct MTU size
+	_, err := m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		doc.OpenShiftCluster.Properties.NetworkProfile.MTUSize = mtuSize
+		return nil
+	})
+	return err
+}
+
+// EnsureMTUSize ensures that an existing cluster object has the MTUSize field defined
+func (m *manager) EnsureMTUSize(ctx context.Context) error {
+
+	// Cluster needs MTUSize field patched
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.MTUSize == 0 {
+		// Get appropriate MTU size
+		mtuSize := api.MTU3900
+
+		// If a single MachineConfig is present we know the cluster has a custom MTU
+		_, err := m.mcocli.MachineconfigurationV1().MachineConfigs().Get(ctx, "99-master-mtu", metav1.GetOptions{})
+		if err != nil {
+			// If we can't find a MachineConfig this cluster never had a custom MTU on install, set to default
+			if kerrors.IsNotFound(err) {
+				mtuSize = api.MTU1500
+			} else {
+				return err
+			}
+		}
+		// Patch the cluster object with correct MTU size
+		_, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
```

Comment:
```suggestion
		m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
```
you want to set the doc with the updated value because it's not dynamically read before we set settings.  So the setting can be lost if the doc isn't updated again.